syntax = "proto3";

package orchestrator;

// Task request from Go
message TaskRequest {
  string engine_name = 1;
  string task_id = 2;
  string input_data = 3; // JSON encoded
}

// Task response back to Go
message TaskResponse {
  string task_id = 1;
  string output_data = 2; // JSON encoded
  string status = 3; // "success" or "error"
}

// gRPC service
service Orchestrator {
  rpc SubmitTask(TaskRequest) returns (TaskResponse);
}
